src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -439,9 +434,16 @@ internal static DkmCustomUIVisualizerInfo[] GetDebuggerCustomUIVisualizerInfo(th
             return result;
         }
 
-        private static string EvaluateDebuggerDisplayString(DkmClrValue value, string str)
+        internal static void EvaluateDebuggerDisplayStringAsync(this DkmClrValue value, DkmWorkList workList, DkmInspectionContext inspectionContext, DkmClrType targetType, string str, CompletionRoutine<string> completionRoutine)
         {
-            return str == null ? null : value.EvaluateDebuggerDisplayString(str);
+            if (str == null)
+            {
+                completionRoutine(null);
```

Comment:
Yeah--we only ever pass our own completion routine here (never one of the ones that Concord supplies).